### ISSUE TITLE
BlobTrigger fails to process a blob

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-functions-dotnet-worker#1842]
This looks like a bug with Azure Functions Isolated Worker SDK.

### Observed issue

The setup is a blob-triggered function below.

```
[Function(nameof(BlobConcierge))]
public async Task Run([BlobTrigger("inspections/{blobName}", 
        Connection = "LossControl360:StorageConnectionString")] BlobInspection blobInspection,
        string blobName, FunctionContext context)
{
    var logger = context.GetLogger(nameof(BlobConcierge));

    // logic that's never invoked
}
```

When a new blob under the `inspections` container is dropped, the function throws.

`Error converting 1 input parameters for Function 'BlobConcierge': Cannot convert input parameter 'blobInspection' to type 'Tva.Endpoints.LossControl360.BlobInspection' from type 'Microsoft.Azure.Functions.Worker.Grpc.Messages.GrpcModelBindingData'. Error:System.InvalidOperationException: Blob storage connection configuration 'StorageConnectionString' does not exist. Make sure that it is a defined App Setting.`
<details><summary>Full stacktrace</summary>
   at Microsoft.Azure.Functions.Worker.BlobStorageBindingOptionsSetup.Configure(String name, BlobStorageBindingOptions options) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\Config\BlobStorageBindingOptionsSetup.cs:line 44
   at Microsoft.Extensions.Options.OptionsFactory`1.Create(String name)
   at Microsoft.Extensions.Options.OptionsMonitor`1.<>c__DisplayClass10_0.<Get>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location ---
   at System.Lazy`1.CreateValue()
   at Microsoft.Extensions.Options.OptionsCache`1.GetOrAdd(String name, Func`1 createOptions)
   at Microsoft.Extensions.Options.OptionsMonitor`1.Get(String name)
   at Microsoft.Azure.Functions.Worker.BlobStorageConverter.CreateBlobContainerClient(String connectionName, String containerName) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\BlobStorageConverter.cs:line 253
   at Microsoft.Azure.Functions.Worker.BlobStorageConverter.ConvertModelBindingDataAsync(Type targetType, BlobBindingData blobData) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\BlobStorageConverter.cs:line 118
   at Microsoft.Azure.Functions.Worker.BlobStorageConverter.ConvertFromBindingDataAsync(ConverterContext context, ModelBindingData modelBindingData) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\BlobStorageConverter.cs:line 63 
</details>

The setting is found in the Functions App Configuration as `LossControl360__StorageConnectionString`.
<img width="196" alt="image" src="https://github.com/Azure/azure-functions-dotnet-worker/assets/1309622/a8773c83-1731-40d1-9389-96ebde4f2c4e">

To workaround this issue, I had to add a new `StorageConnectionString` setting to the Functions App Configuration. 

### Additional information

- .NET 6 / C#
- Relevant Nuget Packages
  - Azure.Storage.Blobs 12.17.0
  - Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs 6.0.1
  - Microsoft.Azure.Functions.Worker.Sdk 1.14.0
  - Microsoft.Azure.Functions.Worker 1.19.0
  - Microsoft.Azure.Functions.Worker.Extensions.Http 3.1.0
  - Microsoft.ApplicationInsights.WorkerService 2.21.0
  - Microsoft.Azure.Functions.Worker.ApplicationInsights 1.0.0-preview5

The failure started recently after updating NuGet packages.

**Comment by SeanFeldman** [comment]
Possibly duplicate of #1826.
@liliankasem

**Comment by liliankasem** [comment]
1826 is specifically an issue with FSharp, I don't think they're related. Likely the way we're resolving connections doesn't seem to work with `:` or` __`. The switch to AzureClientFactory might fix this (#1566) but will do some digging.

**Comment by liliankasem** [comment]
Issue has been addressed and should be in the new blob release:

- https://github.com/Azure/azure-functions-dotnet-worker/releases/tag/storage-extension-6.2.0
